.evergreen/prepare-crypt-shared-lib.sh

```diff
@@ -30,5 +30,5 @@ else
     CRYPT_SHARED_LIB_PATH=$(cygpath -m $CRYPT_SHARED_LIB_PATH)
   fi
   echo "CRYPT_SHARED_LIB_PATH: $CRYPT_SHARED_LIB_PATH"
-  export CRYPT_SHARED_LIB_PATH=${CRYPT_SHARED_LIB_PATH}
+  echo "export CRYPT_SHARED_LIB_PATH=$CRYPT_SHARED_LIB_PATH" >crypt_shared.sh
```

Comment:
drivers-evergreen-tools scripts must be run in bash, but I use zsh locally.  That presents a conundrum - this script relies on bash functions from drivers-evergreen-tools, but run locally for me, this runs in zsh, which causes the script to crash.  So I instead _always_ run it in bash, and export the path to a file which I can source elsewhere.

Unfortunately, adding fish support seemed like too much work so if fish users want it, they'll need to modify the tooling in drivers-evergreen-tools and our own tooling too.

Reply:
This is why people generally prefer to just put `#!/bin/bash` at the start of their scripts + `chmod +x` instead of manually invoking it as `bash <filename>` – might be time to do that here 🙂

Reply:
Does that work when sourcing files too?  I tried that locally but get errors when running with zsh.  Specifically, after invoking `prepare-crypt-shared` with bash, bash exits with:

```
get_distro:.:4: no such file or directory: ./get-distro.sh
```

get_distro gets sourced from drivers-evergreen-tools in the following line:

```
. $DRIVERS_TOOLS/.evergreen/download-mongodb.sh
```

Reply:
Oh, you're right about the actual problem here.  But that's because the script uses BASH_SOURCE, which isn't defined in zsh.

> No, the point of sourcing is that it runs in the same shell instance as the parent 🙂 But that just makes it more desirable to standardize on bash, right?

Are you proposing we drop support for zsh then and simply source everything?

Reply:
```fish
exec bash -c "source some-bash-setup.sh; exec fish"
```

sure... it's a new shell but at least you're back in fish ><>

Reply:
> Are you proposing we drop support for zsh then and simply source everything?

What does "drop support" actually mean here? You'd still be able to run all the `.evergreen/....sh` scripts _from_ any shell you like, really

Reply:
Our local testing is currently set up to run our tests from these scripts.  It's intended that instead you source the credential loading scripts and then run the tests normally

Reply:
Well yeah, then I'd maybe just add a shorthand for what @nbbeeken wrote above ... if the only thing you need is env vars that are output of the credential loading scripts, and not shell functions that need to be shared

Reply:
I'm not following the suggestion - would what Neal suggested be something we use to exec these scripts or would that live somewhere in the scripts I've modified?

Do these changes block the PR?  If we want to make running these tests smoother for fish users, I'd suggest a follow-up PR

Reply:
I'm personally fine breaking that out into another ticket.

Reply:
> would what Neal suggested be something we use to exec these scripts or would that live somewhere in the scripts I've modified?

Neal's suggestion is that if you want to run scripts with a different shell, you could use bash only for setting up the environment variables but then keep using your existing shell

Reply:
If that is a path we wish to explore, I filed NODE-6700.  None of our scripts (not just the FLE scripts) work for fish right now and adapting them to work with other shells is more work than I'd like to do in this PR.

Reply:
I didn't expect any changes here or later, feel free to close that ticket unless there's something you'd like to do to help unblocking zsh _source_-ing

I was pointing out I can still use fish (or any shell) locally if I source these scripts in bash and _then_ launch fish. I posted a shorthand to do it all in one command which is easy enough for me to keep an alias of in my shell's config. 

🤔 Maybe I can do something clever by diffing the env vars that exist before and after launching bash and source all the new ones to the current fish instance to avoid launching a new shell, but I can do that on my own time :)